web/packages/build/jest/jest-environment-patched-jsdom.js

```diff
@@ -80,6 +80,17 @@ export default class PatchedJSDOMEnvironment extends JSDOMEnvironment {
         return result;
       };
     }
+
+    // If a test actually depends on a working ResizeObserver implementation, call
+    // mockResizeObserver provided by jsdom-testing-mocks.
```

Comment:
sorry I am confused, if test works with `jsdom-testing-mocks`, why do we need to handle global `!global.ResizeObserver`. Are they different thing?

Reply:
`global.ResizeObserver` used to be defined in `web/packages/build/jest/setupTests.ts`, which is not the best place for it as we usually define missing JSDOM globals within this file.

JSDOM emulates browser APIs in Node.js, but it doesn't ship with `ResizeObserver` yet. If you do this in a test…

```
import { mockResizeObserver } from 'jsdom-testing-mocks';

const resizeObserver = mockResizeObserver();
```

…jsdom-testing-mocks will mock the whole API and provide you with ways to manipulate it. However, not all tests need that, so for those tests we just provide a simple fake `ResizeObserver` that doesn't do anything.

Reply:
Noted, thanks for explaining Rafał.